src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -305,6 +315,86 @@ export class TerminalService implements ITerminalService {
 
 		// Create async as the class depends on `this`
 		timeout(0).then(() => this._instantiationService.createInstance(TerminalEditorStyle, document.head));
+
+		//TODO:@meganrogge is there a better way to do this?
+		this.onDidReceiveProcessId(async () => await this._refreshDescriptionCwd());
+		this.onDidDisposeInstance(async () => await this._refreshDescriptionCwd());
```

Comment:
Dispose instance would remove the tab so refreshing isn't needed?

Reply:
I was thinking that if you have:
Desktop/xterm.js
repos/xterm.js
when the last one of either is deleted, we'd want it to go back to just xterm.js

Reply:
that doesn't seem to be working ATM anyway though